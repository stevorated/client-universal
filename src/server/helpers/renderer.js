import React from 'react'
import { renderToString } from 'react-dom/server'
import { Helmet } from 'react-helmet'
import { StaticRouter } from 'react-router'
import { Provider } from 'react-redux'
import { ApolloProvider } from 'react-apollo'
import serialize from 'serialize-javascript'
import { ServerStyleSheet } from 'styled-components'
import Layout from '../../shared/Routes/Layout'
import { config } from '@fortawesome/fontawesome-svg-core'
config.autoAddCss = false

const { API_BASE } = process.env
const linkToFavicon = `${API_BASE}/public_images/favicon.png`
export default function renderer(req, store, client, context) {

  const App = (
    <Provider store={store}>
      <ApolloProvider client={client}>
        <StaticRouter location={req.url} context={context}>
          <Layout />
        </StaticRouter>
      </ApolloProvider>
    </Provider>
  )

  const sheet = new ServerStyleSheet()
  const content = renderToString(sheet.collectStyles(App))
  const styles = sheet.getStyleTags()
  
  const helmet = Helmet.renderStatic()

  return `
  <!DOCTYPE html>
    <html>
      <head>
      <meta charset="UTF-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      <meta name="840430c5e70010722d679edee63ea147c19430e1" content="e73a6ac6dc339154d6038998ef2574d65b12bf5e" />
      <script type="text/javascript">
        WebFontConfig = {
          google: { families: [ 'Sigmar One::latin' ] }
        };
        (function() {
          var wf = document.createElement('script');
          wf.src = ('https:' == document.location.protocol ? 'https' : 'http') +
            '://ajax.googleapis.com/ajax/libs/webfont/1/webfont.js';
          wf.type = 'text/javascript';
          wf.async = 'true';
          var s = document.getElementsByTagName('script')[0];
          s.parentNode.insertBefore(wf, s);
        })(); </script>
      <link href="https://fonts.googleapis.com/css?family=Sigmar+One&display=swap" rel="stylesheet" />
      <link href="https://use.fontawesome.com/releases/v5.9.0/css/svg-with-js.css" rel="stylesheet" />
      ${helmet.title.toString()}
      ${helmet.meta.toString()}
      <link rel="icon" href="${linkToFavicon}" type="image/png"/>
      <link rel="stylesheet" href="/styles/main.css"/>
      ${styles}
      </head>
      <body>
        <div id="root">${content}</div>
        <div id="portal"></div>
        <script>
          window.__INITIAL_STATE__ = ${serialize(store.getState())}
        </script>
        <script>
          window.__APOLLO_STATE__ = ${serialize(client.extract())}
        </script>
        <script src="/bundle.js"></script>
      </body>
    </html>
  
  `
}
