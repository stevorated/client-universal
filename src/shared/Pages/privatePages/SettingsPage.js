import React, { Component } from 'react'
import { Row, Col, Button } from 'reactstrap'
import styled from 'styled-components'
import { connect } from 'react-redux'
import {
  HelmetComponent,
  UserSettingsFormContainer,
  ChangePasswordContainer
  // ResetPasswordRequestForm
} from '../../Components'
import { updateCurrentUser } from '../../Store/actions'
import requireAuth from '../../HOC/requireAuth'
import checkLoggedIn from '../../HOC/checkLoggedIn'
import Menu from '../../Routes/Menu'

import { mediaQueries } from '../../Utils'
import { Line } from '../../Elements'

export class SettingsPage extends Component {
  constructor(props) {
    super(props)
    this.title = 'Settings'
    this.state = {
      fname: this.props.auth.fname,
      lname: this.props.auth.lname,
      username: this.props.auth.username,
      bio: this.props.auth.bio,
      fnameActive: false,
      lnameActive: false,
      usernameActive: false,
      bioActive: false,
      fnameError: null,
      fnameGood: null,
      lnameError: null,
      lnameGood: null,
      usernameError: null,
      usernameGood: null,
      bioError: null,
      bioGood: null,

      password: '',
      newPassword: '',
      newPasswordConfirm: '',

      passwordError: null,
      passwordGood: null,
      passwordWeak: null,
      passwordBetter: null,
      passwordGreat: null,

      newPasswordError: null,
      newPasswordGood: null,
      newPasswordWeak: null,
      newPasswordBetter: null,
      newPasswordGreat: null,

      newPasswordConfirmGood: null,
      newPasswordConfirmError: null,
      newPasswordConfirmActive: false,

      passwordUpdated: false,

      email: '',
      emailGood: null,
      emailError: null,
      showErrors: null

    }
    // console.log(this.state)
  }
  handleChange = newData => {
    this.setState({ ...newData })
  }

  handleState = newData => {
    this.setState(newData)
  }

  handleAction = (type, payload) => {
    switch (type) {
      case 'updateCurrentUser':
        // console.log('fetchMyPosts')
        this.props.updateCurrentUser(payload.data)
        break
      default:
        console.log('unKnownAction', type, payload)
        break
    }
  }

  render() {
    return (
      <Row data-test="mainDiv" className="text-center">
        <HelmetComponent
          data-test="helmet"
          pageTitle={this.title}
          ogTitle={this.title}
        />
        <FloatLeft data-test="leftCol" lg="3">
          <Menu />
        </FloatLeft>
        <Col
          data-test="mainCol"
          lg="9"
          className="offset-lg-3 order-3 order-lg-2 animated fadeIn mt-lg-4 mt-3 px-5"
        >
          <Line className="my-4 mx-3" />
          <h1 className="display-4">Profile Details</h1>
          <Line className="my-4 mx-3" />
          <UserSettingsFormContainer
            handleAction={this.handleAction}
            state={this.state}
            handleChange={this.handleChange}
            handleState={this.handleState}
          />
          <Line className="my-4 mx-3" />
          <h1 className="display-4">Password</h1>
          <Line className="my-4 mx-3" />
          {/* <ForgotPasswordContainer
            className="my-5"
            handleAction={this.handleAction}
            state={this.state}
            handleChange={this.handleChange}
            handleState={this.handleState}
          /> */}
          <ChangePasswordContainer
            className="my-5"
            handleAction={this.handleAction}
            state={this.state}
            handleChange={this.handleChange}
            handleState={this.handleState}
          />
        </Col>
      </Row>
    )
  }
}

function mapStateToProps({ auth }) {
  return { auth }
}

export default {
  component: connect(
    mapStateToProps,
    { updateCurrentUser }
  )(checkLoggedIn(requireAuth(SettingsPage)))
}
const FloatLeft = styled(Col)`
  position: static !important;
  top: 3.5rem;
  left: 0rem;
  ${mediaQueries.lg`
  position: fixed!important;
  `}
`
