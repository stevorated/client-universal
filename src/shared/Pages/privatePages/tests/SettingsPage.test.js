import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import configureStore from 'redux-mock-store'

import { SettingsPage } from '../SettingsPage'
import { findByTestAttr } from '../../../../tests/utils'

const initialState = {}
const mockStore = configureStore()
let component, store, tree

const setupShallowRender = (props = {}) => {
  return shallow(<SettingsPage />)
}

describe('<SettingsPage />', () => {
  beforeEach(() => {
    //creates the store with any initial state or middleware needed
    store = mockStore(initialState)
    component = setupShallowRender()
    tree = renderer.create(
      <BrowserRouter>
        <SettingsPage store={store} />
      </BrowserRouter>
    )
  })
  it('matches Snapshot', () => {
    expect(tree).toMatchSnapshot()
  })
  it('Should render witout errors', () => {
    const mainDiv = findByTestAttr(component, 'mainDiv')
    expect(mainDiv.length).toBe(1)
  })
  it('Should render Cols components witout errors', () => {
    const mainCol = findByTestAttr(component, 'mainCol')
    const leftCol = findByTestAttr(component, 'leftCol')

    expect(mainCol.length).toBe(1)
    expect(leftCol.length).toBe(1)
  })
  it('Should render helmet component witout errors', () => {
    const helmet = findByTestAttr(component, 'helmet')

    expect(helmet.length).toBe(1)
    expect(helmet.props().pageTitle).toBe(component.instance().title)
    expect(helmet.props().ogTitle).toBe(component.instance().title)
  })
})
