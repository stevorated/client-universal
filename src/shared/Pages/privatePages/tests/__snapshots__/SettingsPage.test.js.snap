// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsPage /> matches Snapshot 1`] = `undefined`;
